controllers/AdminController.php

```diff
@@ -301,16 +316,27 @@ public function actionInfo($id)
      */
     public function actionSwitch($id)
     {
-        $user = $this->findModel($id);
+        $old = Yii::$app->user;
+        $session = Yii::$app->session;
+        $key = self::ORIGINAL_USER_SESSION_KEY;
 
-        $old = \Yii::$app->user;
+        if($id == 'original' && $session->has($key)) {
```

Comment:
I don't see any problems. If we set method parameter `$id` to be null by default, we will be able to use route `['/user/admin/switch']` to switch back, and route `['/user/admin/switch', 'id' => 123]` to switch to needed user.